internal/cache/async.go

```diff
@@ -80,20 +82,18 @@ func (ac *asyncClient) doCreate(ctx context.Context, key store.Key) {
 	switch {
 	case err == nil:
 		ac.objectStore.OverrideResourceVersionIfNewer(result)
-	case isRetryableError(err):
-		svc1log.FromContext(ctx).Warn("got retryable error, will retry", svc1log.Stacktrace(err))
-		ac.queue.AddIfAbsent(store.CreateRequest(obj))
 	case isNamespaceTerminating(err):
 		svc1log.FromContext(ctx).Info("can not create object because its namespace is being terminated")
-		ac.objectStore.Delete(key)
+		ac.objectStore.Delete(r.Key)
 	default:
-		logNonRetryableError(ctx, err)
-		ac.objectStore.Delete(key)
+		if !ac.maybeRetryRequest(ctx, r, err) {
```

Comment:
nit: would be a bit easier to read if you extract the conditional to a variable named `didRetry`, then the fact that 
we are calling `maybeRetryRequest` for the side effect would be clearer.